src/Native/System.Net.Security.Native/CMakeLists.txt

```diff
@@ -0,0 +1,36 @@
+project(System.Net.Security.Native)
+
+set(CMAKE_INCLUDE_CURRENT_DIR ON)
+
+add_definitions(-DPIC=1)
+
+if (HAVE_HEIMNTLM_HEADERS)
+   find_library(LIB_HEIMNTLM NAMES heimntlm PATHS /usr/heimdal/lib)
+   include_directories(/usr/heimdal/include)
+
+   if(LIB_HEIMNTLM STREQUAL LIB_HEIMNTLM-NOTFOUND)
+       message(FATAL_ERROR "Cannot find libheimntlm.so and System.Net.Security.Native cannot build without it. Try installing libheimntlm (or the appropriate package for your platform)")
+   endif()
+
+
+   set(NATIVENTLM_SOURCES
+     pal_ntlmapi.cpp
+   )
+
+   add_library(System.Net.Security.Native
+    SHARED
+    ${NATIVENTLM_SOURCES}
+   )
+
+   target_link_libraries(System.Net.Security.Native
+           ${LIB_HEIMNTLM}
+           ${OPENSSL_CRYPTO_LIBRARY}
+   )
+
+   install (TARGETS System.Net.Security.Native DESTINATION .)
+
+elseif (CMAKE_SYSTEM_NAME STREQUAL Linux)
+   message (FATAL_ERROR "cannot find libheim ntlm headers and System.Net.Security.Native cannot build without it. Try installing libheimntlm (or the appropriate package for your platform)")
+else ()
```

Comment:
Well, the two things ended up being a bit orthogonal, but both related to "what are we trying to accomplish here?".  I think `PlatformNotSupportedException` is the way to go if we have to live with this past RTM (including for non-primary platforms), and that the `DllNotFoundException` should mean "you're in a bad state", vs "this build doesn't support NTLM".

The trait/enviroment variable/whatever approach does seem like the best we're going to get for supporting it to be required from official builds and not from developer builds (effectively, a one-off test that asserts that it doesn't get a PlatformNotSupportedException when the official-build-state is detected).

So, I like:
* The PlatformNotSupportedException plan (where it doesn't just wrap DllNotFoundException)
* Making the test that ensures that we don't get that exception on official runs
* Making the tests (which hopefully exist) use [ConditionalFact] to skip when PlatformNotSupportedException would be arising.